MetaMask/metamask-extension
# feat: `externally_connectable` CAIP delivery and enveloping

## **Description**

Implements initial Delivery CAIP specs. 

* Adds `BARAD_DUR` feature flag
* Adds `http://*` and `https://*` wildcards to the `externally_connectable` manifest property behind feature flag
* Enables `externally_connectable` CAIP envelope handling from webpage connections (leaves extension connections as is) behind feature flag
* Bifurcates the EIP1993 and CAIP RPC pipelines
  * CAIP RPC Pipeline always responds with error `CAIP RPC Pipeline not yet implemented.`



[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/25075?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/2528
See: https://github.com/jiexi/CAIPs/pull/1

## **Manual testing steps**

1. `BARAD_DUR=1 yarn start`
2. Use your local extension build on Chrome
3. Go to a webpage
4. In console enter the following, replacing EXTENSION_ID with your local extension ID
```
const EXTENSION_ID = 'nonfpcflonapegmnfeafnddgdniflbnk';
const extensionPort = chrome.runtime.connect(EXTENSION_ID)
extensionPort.onMessage.addListener((msg) => console.log('extensionPort on message', msg))
extensionPort.postMessage({type: 'caip-x', data: {method: 'eth_chainId'}})    
```
6. There should be a log with the error response `CAIP RPC Pipeline not yet implemented.`

## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
